src/server/ability.ts

```diff
@@ -25,6 +25,9 @@ function defineAbilityFor(viewerRoles: Roles[]) {
 
   if (includesRoles(viewerRoles, ['editor', 'publisher'])) {
     can('upload', 'Video')
+    // note: an instructor wouldn't have this ability, so they'd only get the
+    // choice to upload as themselves.
+    can('upload', 'Video.WithInstructor')
```

Comment:
Not sure how best to approach the naming here.

Looking at it from the github PR page, it seems like `Video.WithInstructor` isn't really what I want here. This is a course builder, so maybe something like `can('assignInstructor', 'Course')` or something like that.

Thoughts?

Reply:
for the most part we want to be thinking about this from a CRUD perspective, and complex namespaced rules is a slippery slope to hell ;)

so you'd have something like `can('update', 'Video', ['instructor_id'])` under the `editor` roll, otherwise the instructor dropdown doesn't appear and it'd be hard coded to the current user.

I'd probably just put the boolean in the `disabled` field and preselect the current instructor 

in the future you'd probably have something like `can('update', 'Video', ['title', 'description'], {instructor_id: user.instructor.id})` etc

generally with this group this is mostly cosmetic and we don't need to be ultra secure about enforcing this server side where it'd be a little more complex

This is referred to as [Restricting Field Access in the docs](https://casl.js.org/v5/en/guide/restricting-fields)

Reply:
This is spot on. The attribute-level scoping is a way better way to do this kind of thing.

> generally with this group this is mostly cosmetic and we don't need to be ultra secure about enforcing this server side where it'd be a little more complex

I like that with the Next.js `/api` directory, we can use `ability` in both client-side and server-side contexts.